src/components/Filter.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { Filter } from '../types/Filter';
+
+interface Props {
+  todos: Todo[],
+  filter: Filter,
+  setFilter: (filter: Filter) => void;
+}
+
+export const FilterTodo: React.FC<Props> = ({ todos, filter, setFilter }) => (
+  <footer className="todoapp__footer" data-cy="Footer">
+    <span className="todo-count" data-cy="todosCounter">
+      {`${todos.length} items left`}
+    </span>
+
+    <nav className="filter" data-cy="Filter">
+      <a
+        data-cy="FilterLinkAll"
+        href="#/"
+        className={classNames(
+          'filter__link',
+          { selected: filter === Filter.All },
+        )}
+        onClick={() => setFilter(Filter.All)}
+      >
+        All
+      </a>
```

Comment:
you have repeated common links, so you can make one component with them and just change the props